src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.DotNetWatch;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.Watcher.Tools
+{
+    internal class BlazorWebAssemblyDeltaApplier : IDeltaApplier
+    {
+        private readonly IReporter _reporter;
+
+        public BlazorWebAssemblyDeltaApplier(IReporter reporter)
+        {
+            _reporter = reporter;
+        }
+
+        public ValueTask InitializeAsync(DotNetWatchContext context, CancellationToken cancellationToken)
+        {
+            return default;
+        }
+
+        public async ValueTask<bool> Apply(DotNetWatchContext context, string changedFile, DotNetWatchManagedModuleUpdatesWrapper? updates, CancellationToken cancellationToken)
+        {
+            if (context.BrowserRefreshServer is null)
+            {
+                _reporter.Verbose("Unable to send deltas because the refresh server is unavailable.");
+                return false;
+            }
+
+            if (updates is null)
+            {
+                return true;
+            }
+
+            var payload = new UpdatePayload
+            {
+                Deltas = updates.Value.Updates.Select(c => new UpdateDelta
+                {
+                    ModuleId = c.Module,
+                    ILDelta = c.ILDelta.ToArray(),
+                    MetadataDelta = c.MetadataDelta.ToArray(),
+                }),
+            };
+
+            await context.BrowserRefreshServer.SendJsonSerlialized(payload, cancellationToken);
```

Comment:
Is there a failure case here?

Reply:
Exceptions are swallowed at the callsite.